app/controllers/autocomplete_controller.rb

```diff
@@ -88,18 +88,36 @@ def associated_tags
 
   ## NONCANONICAL TAGS
   def noncanonical_tag
-    search_param = params[:term]
+    search_param = Query.new.escape_reserved_characters(params[:term])
     raise "Redshirt: Attempted to constantize invalid class initialize noncanonical_tag #{params[:type].classify}" unless Tag::TYPES.include?(params[:type].classify)
+
     tag_class = params[:type].classify.constantize
-    render_output(tag_class.by_popularity
-                      .where(["canonical = 0 AND name LIKE ?",
-                              '%' + search_param + '%']).limit(10).map(&:name))
+    one_tag = tag_class.find_by(canonical: false, name: search_param)
+    # Is there a tag which is just right ( this is really for testing )
+    match = if one_tag then [one_tag.name]  else [] end # rubocop:disable Style/OneLineConditional
```

Comment:
We really don't do one-line conditionals like this, hence the cop. Either of these would be preferable, please:

```
    match = if one_tag
              [one_tag.name]
            else
              []
            end

    match = one_tag ? [one_tag.name] : []
 ```
 
 Or, since it sounds like Brian's original suggestion didn't work, perhaps even:
 
 ```
     match = []
     match << one_tag.name unless one_tag.nil?
 ```